lib/bson/types.ex

```diff
@@ -1,5 +1,15 @@
 defmodule BSON.Binary do
   defstruct [binary: nil, subtype: :generic]
+
+  defimpl Inspect do
+    def inspect(%BSON.Binary{binary: value, subtype: :generic}, _opts) do
+      "#BSON.Binary<#{Base.encode16(value, case: :lower)}>"
+    end
+
+    def inspect(%BSON.Binary{binary: value, subtype: subtype}, _opts) do
+      "#BSON.Binary(#{subtype})<#{Base.encode16(value, case: :lower)}>"
```

Comment:
Can we print this as: `#BSON.Binary<#{Base.encode16(value, case: :lower)}, #{subtype}>`